### DISCUSSION TITLE
Can you move the IFormFile interface from package Microsoft.AspNETCore.App to Microsoft.NETCore.App?

### DISCUSSION THREAD

**Discussion by lbgitjp** [dotnet/aspnetcore#54209] [Ideas] [answered]
I have a lot of controller code that can be shared across multiple projects, so I moved them to a public library project. I need them to be used not only in the web project, but can also be used in other projects(e.g winforms),  So I just want them to depend on **Microsoft.NETCore.App**, but there is a problem, there is a controller that uses **IFormFile**, which is in the package **Microsoft.AspNETCore.App,**

Now my public library project must depend on two packages **Microsoft.NETCore.App** and **Microsoft.AspNETCore.App**, which is not good, so I hope you can move **IFormFile** from  package **Microsoft.AspNETCore.App** to package **Microsoft.NETCore.App**
, so that my library project  only needs to depend on the package **Microsoft.NETCore.App**.

Thanks!

**Comment by davidfowl** [comment, marked as answer]
Your public library can depend on just Microsoft.AspNETCore.App. This type is in the Microsoft.AspNetCore namespace and won’t be going into the NETCore.App shared framework.

**Reply by lbgitjp** [reply]
I see, thanks!

**Reply by magierska** [reply]
What would you suggest if the library with `IFormFile` needs to be referenced from Blazor WebAssembly? From what I'm seeing wasm doesn't allow to reference `Microsoft.AspNETCore.App` packages?

**Reply by davidfowl** [reply]
I’d ask why IFormFile would be used by blazor WASM

**Reply by magierska** [reply]
Sorry, a bit more insight. We've got blazor wasm project, that is referencing the Domain project. This domain project contains services which will call external APIs. So from the domain project we wanted to reference shared library with DTOs (containing IFormFile), and use these DTOs in the services.

**Reply by danroth27** [reply]
Yeah, you'll need to refactor the domain project so that the stuff you need in your Blazor WebAssembly app doesn't reference Microsoft.AspNetCore.App. This isn't something we plan to change.
